### PR TITLE
update angular-mocks.js url matching

### PR DESCRIPTION
**Pull request by deodeveloper**
url matching was incorrect and updated. if all query parameters in url are in different order, it should still match

### CODE REVIEW

**Comment by deodeveloper** [comment]
I signed it!

**Comment by petebacondarwin** [comment]
Thanks for this @deodeveloper - before we can land this change it would need to have a unit test to show that it is doing what you expect.

**Comment by deodeveloper** [comment]
For merging with master, updated a test case for the changes related to this the pull request as commented by @petebacondarwin

**Comment by gkalpak** [comment]
There are some stying issues (e.g. see https://travis-ci.org/angular/angular.js/jobs/106143760).
You can also run the ci-checks locally (`grunt ci-checks`) - I believe there will be some more (e.g. whitespace between function parameters) once you fix the semicolon issues :wink:

File: src/ngMock/angular-mocks.js [hunk 1]
```diff
@@ -1646,6 +1646,15 @@ function MockHttpExpectation(method, url, data, headers) {
     }
     return data == d;
   };
+  
+  this.getUrlParams = function (u){
```
Comment (gkalpak) [inline src/ngMock/angular-mocks.js]: Do these two methods (`getUrlParams` and `compareUrl`) need to be public ?

**Comment by gkalpak** [comment]
This doesn't properly support URLs with a hash fragment, but I guess we don't really care, because it doesn't make sense to make requests against URLs with a hash fragment (since hash fragments have no meaning for the server).
(In any case, this PR doesn't make them any less supported than they already were, I just mention it as a possible (probably unnecessary) enhancement.)

**Comment by petebacondarwin** [comment]
@deodeveloper can you address the issues that @gkalpak has highlighted?

**Pull request closed without merging**